types/rlp_marshal.go

```diff
@@ -104,6 +113,10 @@ func (r *Receipt) MarshalRLP() []byte {
 }
 
 func (r *Receipt) MarshalRLPTo(dst []byte) []byte {
+	if r.IsTypedTransaction() {
+		dst = append(dst, byte(r.TransactionType))
+	}
+
```

Comment:
Is there a way to move his into `MarshalRLPTo`? What do you think?

Everywhere else in the codebase `MarshalRLPTo` calls `MarshalRLPTo` with the callback `With`

Reply:
Of the receipt. If it's too much of a hassle, then I'm fine with leaving it

Reply:
No, it's impossible. Because need to add TransactionType before RLP(Receipt). If it's done inside Recipt, TransactionType will be included, like RLP(TransactionType || Receipt)
@zivkovicmilos

Reply:
Thank you for clarifying 🙏